packages/aws-cdk-lib/aws-ec2/test/vpc.test.ts

```diff
@@ -379,6 +379,59 @@ describe('vpc', () => {
 
     });
 
+    test('with only reserved subnets as public subnets, should not create the internet gateway', () => {
```

Comment:
Can you please add a similar test case with a  reserved `PRIVATE_WITH_EGRESS` subnet?